libraries/AP_Motors/AP_MotorsHeli_RSC.cpp

```diff
@@ -418,3 +424,70 @@ float AP_MotorsHeli_RSC::calculate_desired_throttle(float collective_in)
     return throttle;
 
 }
+
+// autothrottle_run - calculate throttle output for governor controlled throttle
+void AP_MotorsHeli_RSC::autothrottle_run()
+{
+    float throttlecurve = calculate_throttlecurve(_collective_in);
+
+    // autothrottle main power loop with governor
+    if (_governor_engage && !_governor_fault) {
+        // calculate droop - difference between actual and desired speed
+        float governor_droop = (_governor_rpm - _rotor_rpm) * get_governor_droop_response();
+        _governor_output = governor_droop + ((throttlecurve - _governor_torque_reference) *  get_governor_ff());
+        if (_rotor_rpm < (_governor_rpm - 2.0f)) {
+            _governor_torque_reference += get_governor_compensator();   // torque compensator
+        } else if (_rotor_rpm > (_governor_rpm + 2.0f)) {
+            _governor_torque_reference -= get_governor_compensator();
+        }
+    // throttle output uses droop + torque compensation to maintain proper rotor speed
+    _control_output = constrain_float((_governor_torque_reference + _governor_output), throttlecurve * get_governor_ff(), 1.0f);
+        // governor and speed sensor fault detection - must maintain Rrpm -3/+2%
+        // speed fault detector will allow a fault to persist for 200 contiguous governor updates
+        // this is failsafe for bad speed sensor or severely mis-adjusted governor
+        if ((_rotor_rpm <= (_governor_rpm * 0.97f)) || (_rotor_rpm >= (_governor_rpm * 1.02f))) {
+            _governor_fault_count += 1.0f;
+            if (_governor_fault_count > 200.0f) {
+                _governor_fault = true;
+                _governor_engage = false;
+                _governor_output = 0.0f;
+                _governor_torque_reference = 0.0f;
+                if (_rotor_rpm >= (_governor_rpm * 1.02f)) {
+                    gcs().send_text(MAV_SEVERITY_WARNING, "Governor Fault: Rotor Overspeed");
+                } else {
+                    gcs().send_text(MAV_SEVERITY_WARNING, "Governor Fault: Rotor Underspeed");
+                }
+            }
+        } else {
+            _governor_fault_count = 0.0f;   // reset fault count if the fault doesn't persist
+        }
+    } else if (!_governor_engage && !_governor_fault) {
+        // torque rise limiter accelerates rotor to the reference speed
+        // this limits the max torque rise the governor could call for from the main power loop
+        if (_rotor_rpm >= (_governor_rpm * 0.5f)) {
+            float torque_limit = (get_governor_torque() * get_governor_torque());
+            _governor_output = (_rotor_rpm / _governor_rpm) * torque_limit;
+            _control_output = constrain_float(throttlecurve + _governor_output, 0.0f, 1.0f);
+            _governor_torque_reference = _control_output;  // increment torque setting to be passed to main power loop
+            if (_rotor_rpm >= (_governor_rpm - 2.0f)) {
```

Comment:
No, it can't be a percentage. Percentage builds in "fudge factor". On a rotor that is turning at 540 rpm +/- 0.5% is +/- 2.7 rpm. On a rotor turning at 2,000 rpm +/- 0.5% is +/- 10 rpm. While the percentage would work ok on the 540 rpm rotor, it would be huge "fudge factor" on the 2,000 rpm rotor.

The governor's torque reference is the equivalent of manifold pressure on a piston engine, torque setting on a turbine. It has to start moving it when the rotor rpm is +/-2 or it don't work. Been there, done that, tried it when the governor was designed over a year ago and it has flown slightly over 1,100 hours before it came to ArduPilot. The existing 4 rpm of hysteresis is to deal with the output of the rpm library not returning exactly the same value every time thru the loop. If it weren't for that it would be zero tolerance instead of +/-2